File: 2824-count-pairs-whose-sum-is-less-than-target/2824-count-pairs-whose-sum-is-less-than-target.go
func countPairs(nums []int, target int) int {
    var answer int
	hashMap := make(map[int]int)
	for _, num := range nums {
		for k ,v := range hashMap {
			if num + k < target {
				answer += v
			}
		}
		hashMap[num]++
	}
	return answer
}

File: 2824-count-pairs-whose-sum-is-less-than-target/2824-count-pairs-whose-sum-is-less-than-target.java
import java.util.*;

class Solution {
    public int countPairs(List<Integer> nums, int target) {
        HashMap<Integer,Integer> hashMap = new HashMap<>();
        int answer = 0;

        for (Integer num : nums) {
            for (HashMap.Entry<Integer,Integer> entry: hashMap.entrySet()) {
                if ( num + entry.getKey() < target) {
                    answer += entry.getValue();
                }
            }
            Integer count = hashMap.getOrDefault(num, 0);
            hashMap.put(num, count+1);
        }

        return answer;
    }
}